lib/cronofy/errors.rb

```diff
@@ -8,6 +8,9 @@ def initialize(message=nil)
     end
   end
 
+  class ArgumentError < CronofyError
```

Comment:
While this is namespaced, I'm a little uncomfortable that it duplicates a well-used Ruby exception name.
As it's not Cronofy specific, I'd be tempted to just raise the default one. The common base class is handy for catching any API error case, whereas this is just a pure breach of method contract and not as Cronofy related.

Reply:
I did wonder about this, was torn between the desire to keep all exceptions to a known base class and using the well known exception. Happy to change in favour of the builtin